### PR TITLE
Add a port index mapper service for sFlow

### PR DESCRIPTION
**Pull request by padmanarayana**
**- Why I did it**

sFlow generates both counter samples and flow samples. While the counter samples contain the ifName, the flow samples have only the interface index - and the index is the Linux ifindex (and not the SONiC port index (as is typically seen in the PORT_TABLE). Any external application (sflow collector  / end user) then has to map the index to port index (that's obtained from IF-MIB, say) which is not intuitive. 

The SAI driver generates the samples using the Linux indexing scheme. While it is possible to address the issue from SAI driver/SDK perspective, we believe providing a mapping service and letting applications handle the mapping may be a more flexible alternative (which could benefit other applications in future).

**- How I did it**

Introduce a port_index_mapper script in sflow docker which will:
1. Use IPDB to construct the initial PORT_INDEX_TABLE table in the STATE DB.
2. Register a callback that will watch for RTM_NEWLINK/RTM_DELLINK and update the table accordingly.

Note that this currently supports only interfaces supported by port_util.

**- How to verify it**

1. Checked that the PORT_INDEX_TABLE gets populated by the script at startup (redis-cli -n 6 keys "PORT_INDEX_TABLE*")
2. Used scripts to add/remove port channels to verify changes to the table entries.

Verify that the index/ifindex mapping matches that of "ip link show".

**- Description for the changelog**

Add a port index mapper service for sFlow

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: dockers/docker-sflow/port_index_mapper.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    
+        # Extract the port index from the interface name
+        index = port_util.get_index_from_str(ifname)
+        if index is None:
+            return
+    
+        _hash = '{}|{}'.format(PORT_INDEX_TABLE_NAME, ifname)
+    
+        if msgtype == RTM_NEWLINK:
+            state_db.set(state_db.STATE_DB, _hash, 'index', str(index))
```
Comment (prsunny) [inline dockers/docker-sflow/port_index_mapper.py:57]: I think its better to put updating STATE DB to a separate function, as it is called from multiple places.

**Comment by padmanarayana** [comment]
retest vsimage please

**Comment by prsunny** [comment]
retest vsimage please

**Comment by padmanarayana** [comment]
retest vsimage please

**Comment by prsunny** [comment]
retest vsimage please

**Comment by padmanarayana** [comment]
https://github.com/github/codeql/issues/3746

**Pull request merged**